extensions/amp-gfycat/0.1/amp-gfycat.js

```diff
@@ -48,35 +75,205 @@ class AmpGfycat extends AMP.BaseElement {
   }
 
   /** @override */
-  layoutCallback() {
-    const gfyid = user().assert(
+  buildCallback() {
+    this.videoid_ = this.getVideoId_();
+
+    this.playerReadyPromise_ = new Promise(resolve => {
+      this.playerReadyResolver_ = resolve;
+    });
+
+    installVideoManagerForDoc(this.element);
+    Services.videoManagerForDoc(this.element).register(this);
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    const placeholder = this.win.document.createElement('amp-img');
+    dev().assert(this.videoid_);
+    const videoid = this.videoid_ || '';
+
+    placeholder.setAttribute('src',
+        'https://thumbs.gfycat.com/' +
+        encodeURIComponent(videoid) + '-poster.jpg');
+    placeholder.setAttribute('layout', 'fill');
+    placeholder.setAttribute('placeholder', '');
+    placeholder.setAttribute('referrerpolicy', 'origin');
+    this.applyFillContent(placeholder);
+
+    return placeholder;
+  }
+
+  /**
+   * @return {string}
+   * @private
+   */
+  getVideoId_() {
+    return user().assert(
         this.element.getAttribute('data-gfyid'),
         'The data-gfyid attribute is required for <amp-gfycat> %s',
         this.element);
-    const noautoplay = this.element.hasAttribute('noautoplay');
+  }
 
-    const iframe = this.element.ownerDocument.createElement('iframe');
-    iframe.setAttribute('frameborder', '0');
+  /** @return {string} */
+  getVideoIframeSrc_() {
+    if (this.videoIframeSrc_) {
+      return this.videoIframeSrc_;
+    }
+
+    const videoid = this.videoid_ || '';
+    let src = 'https://gfycat.com/ifr/' + encodeURIComponent(videoid);
+
+    const params = getDataParamsFromAttributes(this.element);
 
-    let src = 'https://gfycat.com/ifr/' + encodeURIComponent(gfyid);
+    const noautoplay = this.element.hasAttribute('noautoplay');
```

Comment:
as mentioned above, remove if you like AMP to manage autoplay (if not please let me know since there is no point in implementing the interface in that case)